library/X509/Job.php

```diff
@@ -456,32 +458,32 @@ protected function processChain($target, $chain)
                 );
                 $targetId = $this->db->lastInsertId();
             } else {
-                $targetId = $row->id;
+                $targetId = (int) $row->id;
             }
 
             $chainUptodate = false;
 
-            $lastChain = $this->db->select(
-                (new Select())
-                    ->columns(['id'])
-                    ->from('x509_certificate_chain')
-                    ->where(['target_id = ?' => $targetId])
-                    ->orderBy('id', SORT_DESC)
-                    ->limit(1)
-            )->fetch();
-
-            if ($lastChain !== false) {
-                $lastFingerprints = $this->db->select(
-                    (new Select())
-                        ->columns(['c.fingerprint'])
-                        ->from('x509_certificate_chain_link l')
-                        ->join('x509_certificate c', 'l.certificate_id = c.id')
-                        ->where(['l.certificate_chain_id = ?' => $lastChain->id])
-                        ->orderBy('l.order')
-                )->fetchAll();
-
-                foreach ($lastFingerprints as &$lastFingerprint) {
-                    $lastFingerprint = $lastFingerprint->fingerprint;
+            $lastChain = X509CertificateChain::on($this->db)
+                ->columns(['id'])
+                ->filter(Filter::equal('target_id', $targetId))
+                ->orderBy('id', SORT_DESC)
+                ->limit(1)
+                ->first();
+
+            if ($lastChain) {
+                $lastFingerprints = X509Certificate::on($this->db)->utilize('chain');
+                $lastFingerprints
+                    ->columns(['fingerprint'])
+                    ->getSelectBase()
+                    ->where(new Expression(
+                        'certificate_link.certificate_chain_id = %d',
+                        [(int) $lastChain->id]
+                    ))
+                    ->orderBy('certificate_link.order');
+
+                $lastFingerprintsArr = [];
```

Comment:
We have such code here and there and I think it makes sense to add a method for it in our ORM, i.e. `pluck(string ...$columns)`. Could you please open an issue for that?

Reply:
Done ✅